exercises/paasio/tests/paasio.rs

```diff
@@ -187,3 +187,14 @@ read_file(
     ::std::fs::File::open("README.md").expect("readme must be present"),
     |f: &::std::fs::File| f.metadata().expect("metadata must be present").len() as usize
 ));
+
+#[test]
+#[ignore]
+fn read_stats_by_ref_returns_wrapped_reader() {
+    use paasio::ReadStats;
+
+    let input =
+        "Why, sometimes I've believed as many as six impossible things before breakfast".as_bytes();
+    let reader = ReadStats::new(input);
+    assert_eq!(reader.get_ref(), &input);
```

Comment:
This compares equality, but not identity. You may want to try something like this (note that I haven't tested this yet):
```suggestion
    assert_eq!(reader.get_ref(), &input);
    assert_eq!(reader.get_ref() as *const _, &input as *const _); // test that the returned reader is the wrapped reader
```

Reply:
Comparing casts to `*const` failed with the error:

```
---- read_stats_by_ref_returns_wrapped_reader stdout ----
thread 'read_stats_by_ref_returns_wrapped_reader' panicked at 'assertion failed: `(left == right)`
  left: `0x7f0bd21fc600`,
 right: `0x7f0bd21fc5f0`', tests/paasio.rs:198:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

I had better luck with

```rust
assert_eq!(reader.get_ref().as_ptr(), input.as_ptr()); // test that the returned reader is the wrapped reader
```

But it is very hard to test that the test is effective.

Even when I use 2 different ways of constructing the slices such as:

```rust
    let input = &['W' as u8];
    let reader = ReadStats::new("W".as_bytes());
    assert_eq!(reader.get_ref().as_ptr(), input.as_ptr()); // test that the returned reader is the wrapped reader
```

it passes. I'm guessing the rust compiler works out that both slices have the same content and collapses them both into the same memory.

To answer your questions:

* I don't know if it's a good idea, but I don't think it's necessary. `get_ref` returning a reference to the original wrapped reader is the easiest solution to this test. Doing anything else would require a lot of effort from the person solving the exercise, since `get_ref` is a generic method and can return many different types. Tests can almost always be bypassed if someone puts in enough effort to do so.
* I'm not sure. It's not syntax I'm familiar with. Casting to `*const` is what I found when googling "rust compare identity". But it didn't pass on my laptop.
* Both casting to `*const` and using `as_ptr` behaved the same for me with and without `--release` specified.